integration-tests/src/test/java/oracle/weblogic/kubernetes/ItRetryImprovements.java

```diff
@@ -0,0 +1,283 @@
+    String opLogSevereErrRegex = new StringBuffer(".*SEVERE")
+        .append(".*WebLogicCredentials.*\\s*secret.*")
+        .append(wlSecretName)
+        .append(".*not\\s*found\\s*in\\s*namespace\\s*.*")
+        .append(domainNamespace).toString();
+
+    testUntil(() -> findStringInOperatorLog(opLogSevereErrRegex), logger, "SEVERE error found in Operator log");
+  }
+
+  /**
+   * Create a domain-in-image domain before the secret for admin credentials is created.
+   * Verify that retry stops after the issue is fixed and the domain starts successfully.
+   */
+  @Test
+  @DisplayName("Verify that retry stops after the issue is fixed and the domain starts successfully.")
+  void testRetryStoppedAfterIssueFixed() {
+    // verify that the operator starts retrying when a SEVERE error occurs
+    Long failureRetryLimitMinutes = Long.valueOf("5");
+    createDomainWithoutWlsSecret(failureRetryLimitMinutes);
+
+    String retryOccurRegex = new StringBuffer(".*WebLogicCredentials.*\\s*secret.*")
+        .append(wlSecretName)
+        .append(".*not\\s*found\\s*in\\s*namespace\\s*.*")
+        .append(domainNamespace)
+        .append(".*Will\\s*retry\\s*next\\s*at.*and\\s*approximately\\s*every\\s*")
+        .append(FAILURE_RETRY_INTERVAL_SECONDS)
+        .append("\\s*seconds\\s*afterward\\s*until.*\\s*if\\s*the\\s*failure\\s*is\\s*not\\s*resolved").toString();
+
+    testUntil(() -> findStringInDomainStatusMessage(retryOccurRegex), logger, "retry occurs as expected");
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    createSecretWithUsernamePassword(wlSecretName, domainNamespace, ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT);
+
+    // wait for the domain to exist
+    logger.info("Checking for domain custom resource in namespace {0}", domainNamespace);
+    testUntil(
+        domainExists(domainUid, DOMAIN_VERSION, domainNamespace),
+        logger,
+        "domain {0} to be created in namespace {1}",
+        domainUid,
+        domainNamespace);
+
+    // check that admin service/pod exists in the domain namespace
+    String adminServerPodName = domainUid + "-" + ADMIN_SERVER_NAME_BASE;
+    String managedServerPodNamePrefix = domainUid + "-" + MANAGED_SERVER_NAME_BASE;
+    logger.info("Checking that admin service/pod {0} exists in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace);
+
+    for (int i = 1; i <= replicaCount; i++) {
+      String managedServerPodName = managedServerPodNamePrefix + i;
+
+      // check that ms service/pod exists in the domain namespace
+      logger.info("Checking that clustered ms service/pod {0} exists in namespace {1}",
+          managedServerPodName, domainNamespace);
+      checkPodReadyAndServiceExists(managedServerPodName, domainUid, domainNamespace);
+    }
+  }
+
+  private boolean findStringInDomainStatusMessage(String regex) {
```

Comment:
It is a candidate for a method in a Domain Utility class.

Reply:
moved